cas-server-webapp-config/src/main/java/org/apereo/cas/config/CasAuditTrailConfiguration.java

```diff
@@ -67,12 +65,14 @@
     /**
      * Audit trail management aspect audit trail management aspect.
      *
+     * @param centralAuthenticationService
      * @return the audit trail management aspect
      */
+    @Autowired
     @Bean(name = "auditTrailManagementAspect")
-    public AuditTrailManagementAspect auditTrailManagementAspect() {
+    public AuditTrailManagementAspect auditTrailManagementAspect(CentralAuthenticationService centralAuthenticationService) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Parameter centralAuthenticationService should be final.](https://www.codacy.com/app/mmoayyed/cas/file/2639516913/issues/source?bid=3256115&fileBranchId=3321105#l73)